libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -933,6 +933,25 @@ string YulUtilFunctions::calldataArrayIndexAccessFunction(ArrayType const& _type
 	});
 }
 
+string YulUtilFunctions::calldataArrayIndexRangeAccess(ArrayType const& _type)
+{
+	solAssert(_type.dataStoredIn(DataLocation::CallData), "");
```

Comment:
You should also assert that ``_type.isDynamicallySized()``.
At the moment for statically sized types we don't have slices (it's a type checker error I think). If we did, then those wouldn't have ``length`` on the stack, so the function would have to behave differently for that case - so for now we just assert.

Reply:
Fixed